packages/kbn-ftr-common-functional-services/services/saml_auth/saml_auth_provider.ts

```diff
@@ -1,82 +1,60 @@
 /*
  * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
  * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
  */
 
-import { ServerlessProjectType, SERVERLESS_ROLES_ROOT_PATH } from '@kbn/es';
 import { SamlSessionManager } from '@kbn/test';
-import { readRolesDescriptorsFromResource } from '@kbn/es';
-import { resolve } from 'path';
-import { Role } from '@kbn/test/src/auth/types';
-import { isServerlessProjectType } from '@kbn/es/src/utils';
 import expect from '@kbn/expect';
-import { FtrProviderContext } from '../../functional/ftr_provider_context';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { resolve } from 'path';
+import { FtrProviderContext } from '../ftr_provider_context';
+import { getAuthRoleProvider } from './get_auth_provider';
+import { InternalRequestHeader } from './default_request_headers';
 
 export interface RoleCredentials {
   apiKey: { id: string; name: string };
   apiKeyHeader: { Authorization: string };
   cookieHeader: { Cookie: string };
 }
 
-export function SvlUserManagerProvider({ getService }: FtrProviderContext) {
+export async function SamlAuthProvider({ getService }: FtrProviderContext) {
```

Comment:
Temporarily making it `async` because for stateful SAML setup we need to create `admin` role and apply roles mapping. Without it we can't run the tests.
As soon as this logic is moved to `@kbn/es`, we can make service synchronous function again.